recipe_scrapers/onehundredonecookbooks.py

```diff
@@ -17,27 +16,56 @@ def author(self):
         return self.schema.author()
 
     def title(self):
-        return self.soup.find("h1").get_text()
+        return self.soup.find("h2").get_text()
 
     def total_time(self):
         return self.schema.total_time()
 
     def yields(self):
-        data = self.soup.find_all("p", limit=3, recursive=False)[-1].get_text()
-        extraction = re.search("([0-9]+) servings", data)
-        return extraction.group(1) if extraction else None
+        header = self.soup.find("div", class_="cb101-recipe-time-header")
+        if header and "Serves" in header.text:
+            data = self.soup.find("div", class_="cb101-recipe-time").text.strip()
+            total_yields = data.split()[0]
+            return get_yields(total_yields)
 
     def image(self):
         return self.schema.image()
 
     def ingredients(self):
-        ingredients = self.soup.find("blockquote").p.stripped_strings
-        return list(ingredients)
+        ingredients = []
 
-    def instructions(self):
-        return self.soup.find_all("p", limit=2, recursive=False)[1].get_text(
-            "\n", strip=True
+        ingredient_items = self.soup.select(
+            ".cb101-recipe-ingredients li.cb101-recipe-ingredient"
         )
 
-    def ratings(self):
-        return None
+        for item in ingredient_items:
+            amount_element = item.select_one(".cb101-recipe-ingredient-amount")
+            unit_element = item.select_one(".cb101-recipe-ingredient-unit")
+            name_element = item.select_one(".cb101-recipe-ingredient-name")
+
+            amount = amount_element.get_text(strip=True) if amount_element else ""
+            unit = unit_element.get_text(strip=True) if unit_element else ""
+            name = name_element.get_text(strip=True)
+
+            if amount and unit:
+                ingredient = f"{amount} {unit} {name}".strip()
+            else:
+                ingredient = f"{amount}{unit} {name}".strip()
+
+            ingredients.append(ingredient)
```

Comment:
Probably me just [code golfing](https://en.wikipedia.org/wiki/Code_golf), but I would suggest:

```suggestion
            amount = amount_element.text if amount_element else ""
            unit = unit_element.text if unit_element else ""
            name = name_element.text

            quantity = f"{amount} {unit}".strip()
            ingredient = f"{quantity} {name}".strip()
            ingredients.append(ingredient)
```

(concatenate each pair of text components, and then remove all surrounding whitespace in a single operation on that result)